server/schedule/placement/rule_manager.go

```diff
@@ -169,8 +173,13 @@ func (m *RuleManager) SetRule(rule *Rule) error {
 	old := m.rules[rule.Key()]
 	m.rules[rule.Key()] = rule
 
-	if err = m.store.SaveRule(rule.StoreKey(), rule); err != nil {
-		// restore
+	ruleList, err := buildRuleList(m.rules)
+	if err == nil {
+		err = m.store.SaveRule(rule.StoreKey(), rule)
+	}
+
+	// restore
+	if err != nil {
```

Comment:
else?

Reply:
cannot use `else`, `err` may become not nil in if

Reply:
got it.